google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1alpha2/JsonToProtoMessage.java

```diff
@@ -0,0 +1,322 @@
+              convertJsonToProtoMessageImpl(
+                  fieldDescriptor.getMessageType(),
+                  json.getJSONObject(exactJsonKeyName),
+                  currentScope,
+                  /*topLevel =*/ false,
+                  allowUnknownFields));
+          break;
+      }
+    } catch (JSONException e) {
+      throw new IllegalArgumentException(
+          String.format(
+              "JSONObject does not have a %s field at %s.",
+              FieldTypeToDebugMessage.get(fieldDescriptor.getType()), currentScope));
+    }
+  }
+
+  /**
+   * Fills a repeated protoField with the json data.
+   *
+   * @param protoMsg The protocol buffer message being constructed
+   * @param fieldDescriptor
+   * @param json If root level has no matching fields, throws exception.
+   * @param exactJsonKeyName Actual key name in JSONObject instead of lowercased version
+   * @param currentScope Debugging purposes
+   * @param allowUnknownFields Ignores unknown JSON fields.
+   * @throws IllegalArgumentException when JSON data is not compatible with proto descriptor.
+   */
+  private static void fillRepeatedField(
+      DynamicMessage.Builder protoMsg,
+      FieldDescriptor fieldDescriptor,
+      JSONObject json,
+      String exactJsonKeyName,
+      String currentScope,
+      boolean allowUnknownFields)
+      throws IllegalArgumentException {
+
+    JSONArray jsonArray;
+    try {
+      jsonArray = json.getJSONArray(exactJsonKeyName);
+    } catch (JSONException e) {
+      throw new IllegalArgumentException(
+          "JSONObject does not have a array field at " + currentScope + ".");
+    }
+    java.lang.Object val;
+    int index = -1;
+    try {
+      switch (fieldDescriptor.getType()) {
+        case BOOL:
+          for (int i = 0; i < jsonArray.length(); i++) {
+            index = i;
+            protoMsg.addRepeatedField(fieldDescriptor, new Boolean(jsonArray.getBoolean(i)));
+          }
+          break;
+        case BYTES:
+          for (int i = 0; i < jsonArray.length(); i++) {
+            index = i;
+            protoMsg.addRepeatedField(fieldDescriptor, jsonArray.getString(i).getBytes());
+          }
+          break;
+        case INT64:
+          for (int i = 0; i < jsonArray.length(); i++) {
+            index = i;
+            val = jsonArray.get(i);
+            if (val instanceof Integer) {
+              protoMsg.addRepeatedField(fieldDescriptor, new Long((Integer) val));
+            } else if (val instanceof Long) {
+              protoMsg.addRepeatedField(fieldDescriptor, new Long((Long) val));
+            } else {
+              throw new JSONException("");
+            }
+          }
+          break;
+        case INT32:
+          for (int i = 0; i < jsonArray.length(); i++) {
+            index = i;
+            val = jsonArray.get(i);
+            if (val instanceof Integer) {
+              protoMsg.addRepeatedField(fieldDescriptor, new Integer((Integer) val));
```

Comment:
can new Integer((Integer)  val) just be new Integer(val) or (Integer) val?

Reply:
Yes I'll change that now.